### PR TITLE
dnscrypt-proxy 1.9.4

### PR DESCRIPTION
**Pull request by justinmayer**
This updates dnscrypt-proxy to version 1.9.4. It also moves settings
from the plist to a configuration file, as explained below.

Configuring DNSCrypt settings via arguments in the plist meant that
changes were overwritten on each `brew upgrade`. Since DNSCrypt recently
added support for configuration files, this formula now uses the
configuration file in order to allow for persistent DNSCrypt settings.

----------------

* [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
* [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
* [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
* [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by justinmayer** [comment]
@ilovezfs / @jedisct1: This pull request updates Homebrew's `dnscrypt-proxy` formula to utilize the configuration file introduced in version 1.8+. Comments welcome.

**Comment by ilovezfs** [comment]
There's now 1.9.2 if you'd like to refresh this as a version bump.

**Comment by justinmayer** [comment]
@ilovezfs: Wonderful. Thanks for the heads-up. Will do that shortly.

**Comment by justinmayer** [comment]
Formula for dnscrypt-proxy 1.9.2 has been updated and is ready for review.

File: Formula/dnscrypt-proxy.rb [hunk 1]
```diff
@@ -24,9 +24,18 @@ class DnscryptProxy < Formula
   depends_on "ldns" => :recommended
 
   def install
+    # Configure an initial resolver
+    inreplace "dnscrypt-proxy.conf", "ResolverName please-change-the-resolver-name-in-the-config-file", "ResolverName dnscrypt.eu-dk"
```
Comment (MikeMcQuaid) [inline Formula/dnscrypt-proxy.rb]: Can you split this (and those below) onto 2-3 lines for readability? Thanks!

**Review by MikeMcQuaid — DISMISSED** [review]

**Comment by ilovezfs** [comment]
And it's at 1.9.3 now! This release schedule is breakneck.

**Comment by justinmayer** [comment]
Indeed. That said, can't really complain about early+often releases!

Traveling this week, but I'll update this PR over the weekend.

**Comment by justinmayer** [comment]
@MikeMcQuaid: I made the requested changes (one of which was recently rendered irrelevant). I also updated the formula to yesterday's 1.9.4 release. Would you please review and merge if all looks good here?

**Comment by ilovezfs** [comment]
On Yosemite, `dnscrypt-update-resolvers` fails with
```
yosemitevm:~ brew$ /usr/local/opt/dnscrypt-proxy/bin/dnscrypt-update-resolvers  
+ RESOLVERS_UPDATES_BASE_URL=https://download.dnscrypt.org/dnscrypt-proxy
+ RESOLVERS_LIST_BASE_DIR=/usr/local/Cellar/dnscrypt-proxy/1.9.1/share/dnscrypt-proxy
+ RESOLVERS_LIST_PUBLIC_KEY=RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3
+ curl -v -L --max-redirs 5 -4 -m 30 --connect-timeout 30 -s https://download.dnscrypt.org/dnscrypt-proxy/dnscrypt-resolvers.csv
*   Trying 37.59.238.213...
* Connected to download.dnscrypt.org (37.59.238.213) port 443 (#0)
* SSL peer handshake failed, the server most likely requires a client certificate to connect
* Closing connection 0
```

**Comment by justinmayer** [comment]
@ilovezfs: Saw that. Could be due to Yosemite's old version of OpenSSL. Not sure how to resolve that one.

**Comment by ilovezfs** [comment]
We can do any of these:
1. `depends_on :macos => :el_capitan` and drop Yosemite support
2. `depends_on curl => "with-openssl"` and have dnscrypt-update-resolvers use that if building with the minisign option and on Yosemite
3. remove the minisign option on Yosemite

If 2, the minisign option should probably be optional not recommended on Yosemite.

My initial inclination is 1.

@MikeMcQuaid thoughts?

**Comment by MikeMcQuaid** [comment]
> depends_on curl => "with-openssl" and have dnscrypt-update-resolvers use that if building with the minisign option and on Yosemite
> remove the minisign option on Yosemite
> If 2, the minisign option should probably be optional not recommended on Yosemite.

I'd prefer one of these but don't have a preference as to which.

**Comment by ilovezfs** [comment]
Since we were talking the other day about possibly auditing out all  `=> "with-foo"` options at some point, if not 1, then probably 3 is best.

**Comment by justinmayer** [comment]
@ilovezfs: What would that look like for this formula?

**Comment by ilovezfs** [comment]
```
depends_on "minisign" => :recommended if MacOS.version > :yosemite
```
or
```
depends_on "minisign" => :recommended if MacOS.version >= :el_capitan
```

**Comment by MikeMcQuaid** [comment]
@ilovezfs I defer to you here.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by justinmayer** [comment]
@ilovezfs: Thanks for the suggested modifications. I just pushed those changes and am awaiting test results.

**Comment by ilovezfs** [comment]
@justinmayer actually I already merged that change

**Comment by justinmayer** [comment]
Tests are passing now. Looks like all is ready to merge. 👍

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Shipped! Thanks for the good work here.

**Comment by justinmayer** [comment]
Likewise! Nice team effort. 🎉

**Comment by jedisct1** [comment]
Good job everyone! 👍

**Pull request closed without merging**